lib/atecc508a/transport/cache.ex

```diff
@@ -15,60 +17,71 @@ defmodule ATECC508A.Transport.Cache do
   end
 
   @doc """
-  Check if the specified request is in the cache
+  Check the cache for the request
   """
-  @spec get(GenServer.server(), binary()) :: binary() | nil
+  @spec get(GenServer.server(), binary()) :: response() | nil
   def get(pid, request) do
     GenServer.call(pid, {:get, request})
   end
 
   @doc """
-  Save a response back to the cache
+  Selectively cache responses
   """
-  @spec put(GenServer.server(), binary(), any()) :: any()
-
-  # Cache all reads
-  def put(pid, <<@atecc508a_op_read, _::binary>> = request, response) do
-    GenServer.call(pid, {:put, request, response})
-  end
-
-  # Cache the response to getting a public key
-  def put(pid, <<@atecc508a_op_genkey, 0, _key_id::little-16>> = request, response) do
-    GenServer.call(pid, {:put, request, response})
-  end
-
-  # Don't cache random numbers
-  def put(_pid, <<@atecc508a_op_random, _::binary>>, response), do: response
+  @spec put(GenServer.server(), binary(), response()) :: response()
```

Comment:
```suggestion
  @spec put(GenServer.server(), binary(), response()) :: :ok
```

The spec says the response is returned, but all the functions in the case are actually just returning `:ok`

Reply:
I think GitHub's diff view is misleading. The response is returned after the case.